src/ansys/fluent/core/utils/search.py

```diff
@@ -224,10 +248,429 @@ def inner(tree, path, root_path):
                     next_path = f'{path}.{k}["<name>"]'
                 else:
                     next_path = f"{path}.{k}"
+                type_ = "Object" if isinstance(v, Mapping) else v
+                api_object_names.add(k)
+                if "tui" in next_path:
+                    api_tui_objects.append(f"{next_path} ({type_})")
+                else:
+                    api_objects.append(f"{next_path} ({type_})")
                 if _match(k, word, match_whole_word, match_case):
-                    type_ = "Object" if isinstance(v, Mapping) else v
                     print(f"{next_path} ({type_})")
             if isinstance(v, Mapping):
                 inner(v, next_path, root_path)
 
     inner(api_tree, "", root_path)
+
+    def _write_api_tree_file(
+        api_objects: list,
+        name: Optional[bool] = None,
+        text: Optional[bool] = False,
+        tui: Optional[bool] = False,
+    ):
+        api_tree_file = get_api_tree_file_name(text=text, name=name, tui=tui)
+        api_tree_file.touch()
+        with open(api_tree_file, "w") as file:
+            for api_object in api_objects:
+                file.write(f"{api_object}")
+                file.write("\n")
+
+    text_file_folder = Path(os.path.join(pyfluent.CODEGEN_OUTDIR, "api_tree"))
+    text_file_folder.mkdir(parents=True, exist_ok=True)
+
+    _write_api_tree_file(api_objects=api_objects, text=True)
+    _write_api_tree_file(api_objects=api_tui_objects, tui=True)
+    _write_api_tree_file(api_objects=api_object_names, name=True)
+
+
+def _get_api_object_names():
+    """Get API object names."""
+    return [
+        line.rstrip("\n")
+        for line in open(get_api_tree_file_name(name=True), "r", encoding="utf-8")
+    ]
+
+
+def _print_search_results(queries: list):
+    """Print search results.
+
+    Parameters
+    ----------
+    queries: list
+        List of search string to match API object names.
+    """
+    api_tree = get_api_tree_file_name(text=True)
+    api_tui_tree = get_api_tree_file_name(tui=True)
+    text_files = [api_tree, api_tui_tree]
+    for text_file in text_files:
+        api_objects = [
+            line.rstrip("\n") for line in open(text_file, "r", encoding="utf-8")
+        ]
+        for query in queries:
+            for api_object in api_objects:
```

Comment:
Can this become the outer loop and just iterate over a generator?

Reply:
Yes, we will try this.

Reply:
We are using this in various functions as per the requirement. Therefore it can not become the outer loop.

Reply:
I'm not sure what that happens. Could you please explain when we meet?